modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -262,6 +294,41 @@ private void logExtensionHotUpdate(Extension extension, Extension actual)
         }
     }
 
+    private ConfigurationExpirationMonitor newConfigurationExpirationManager()
+    {
+        Time freq = getConfigurationExpirationFrequency();
+        return newBuilder(extensionRegistry, muleContext)
+                .runEvery(freq.getTime(), freq.getUnit())
+                .onExpired((key, object) -> unregisterConfigurationInstance(key, object))
+                .build();
+    }
+
+    private void unregisterConfigurationInstance(String key, Object object)
+    {
+        try
+        {
+            muleContext.getRegistry().unregisterObject(key);
+        }
+        catch (RegistrationException e)
+        {
+            LOGGER.error(String.format("Could not unregister expired dynamic config of key '%s' and type %s",
+                                       key, object.getClass().getName()), e);
+        }
+    }
+
+    private Time getConfigurationExpirationFrequency()
+    {
+        ExtensionConfig extensionConfig = muleContext.getConfiguration().getExtension(ExtensionConfig.class);
+        if (extensionConfig != null)
```

Comment:
no. Every config can have settings of its own, such as the expiration-policy. But we also need application wide configurations for application wide concerns